addOns/reports/src/main/java/org/zaproxy/addon/reports/sarif/SarifBinaryContentDetector.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.reports.sarif;
+
+import static java.util.Objects.requireNonNull;
+
+import org.parosproxy.paros.network.HttpHeader;
+
+public class SarifBinaryContentDetector {
+
+    public static final SarifBinaryContentDetector DEFAULT = new SarifBinaryContentDetector();
+
+    /**
+     * Detect binary content by inspecting the given HTTP header and the returned normalized
+     * (lowercased) content type. see <a
+     * href="https://www.iana.org/assignments/media-types/media-types.xhtml">IANA Mime type list</a>
+     * for a list of mime types.
+     *
+     * @param header
+     * @return
+     */
+    public boolean isBinaryContent(HttpHeader header) {
+        requireNonNull(header, "Header parameter may not be null!");
+
+        String contentTypeLowerCased = header.getNormalisedContentTypeValue();
+        if (contentTypeLowerCased == null) {
+            // if not set, we assume it is binary
+            return true;
+        }
+
+        boolean isText = contentTypeLowerCased.startsWith("text");
+        isText = isText || contentTypeLowerCased.indexOf("/json") != -1;
+        isText = isText || contentTypeLowerCased.indexOf("/xml") != -1;
+
+        return !isText;
```

Comment:
### Why the multiple lines?
1. debugging is much more simpler when using such structure - just make a breakpoint at the condition you want to inspect and it works - in every IDE
2. Maintainance and Readability

### Why the indexOf?
1. hmm... seems to be a quirk of myself... - after reading implemention of `String.contains` it makes no performance
and "contains" is much more readable and understandable.

So I will  change to "contains" (easier to read). 
Do you want a one-liner or do you find the multiple line solution better?

Reply:
I just figured a single return was more efficient, since they're all or'd as soon as one is true then it goes. This was it actually has to do 3 operations. I understand this is minimal in 2021 and modern processors, but still.

Also I think our normal linter (Sonar) will flag it as well :wink:

Reply:
@kingthorin : I will change this as you like - but only a remark: I don't think there is any difference inside the final byte code. It is  just IMHO more readable. But of course I apply to your code styles.